pkgs/development/compilers/llvm/13/libcxx/default.nix

```diff
@@ -24,15 +32,29 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ cmake python3 ]
     ++ lib.optional stdenv.isDarwin fixDarwinDylibNames;
 
-  cmakeFlags = [
-  ] ++ lib.optional (stdenv.hostPlatform.isMusl || stdenv.hostPlatform.isWasi) "-DLIBCXX_HAS_MUSL_LIBC=1"
+  buildInputs = lib.optionals (!headersOnly) [ libcxxabi ];
+
+  cmakeFlags = [ "-DLIBCXX_CXX_ABI=libcxxabi" ]
+    ++ lib.optional (stdenv.hostPlatform.isMusl || stdenv.hostPlatform.isWasi) "-DLIBCXX_HAS_MUSL_LIBC=1"
     ++ lib.optional (stdenv.hostPlatform.useLLVM or false) "-DLIBCXX_USE_COMPILER_RT=ON"
-    ++ lib.optional stdenv.hostPlatform.isWasm [
+    ++ lib.optionals stdenv.hostPlatform.isWasm [
       "-DLIBCXX_ENABLE_THREADS=OFF"
       "-DLIBCXX_ENABLE_FILESYSTEM=OFF"
       "-DLIBCXX_ENABLE_EXCEPTIONS=OFF"
     ] ++ lib.optional (!enableShared) "-DLIBCXX_ENABLE_SHARED=OFF";
 
+  buildFlags = lib.optional headersOnly "generate-cxx-headers";
+  installTargets = lib.optional headersOnly "install-cxx-headers";
+
+  # At this point, cxxabi headers would be installed in the dev output, which
+  # prevents moveToOutput from doing its job later in the build process.
+  postInstall = lib.optionalString (!headersOnly) ''
```

Comment:
@midchildan do you remember why this was necessary?  What difference would it make having headers installed directly into $dev compared to having moveToOutput move them there later?

Reply:
After `installPhase`, libcxxabi headers get installed to `$dev` while all other headers get installed to `$out`. This causes problems because `moveToOutput` requires the destination directory to be empty beforehand.

```console
❯ cat default.nix
{ pkgs ? import <nixpkgs> {} }:

pkgs.llvmPackages_13.libcxx.overrideAttrs (_: {
  postInstall = "";
})
❯ nix-build
...
-- Installing: /nix/store/v331jri254xij8a9fw8livmvzmfba1hk-libcxx-13.0.1/lib/libc++experimental.a
post-installation fixup
Moving /nix/store/v331jri254xij8a9fw8livmvzmfba1hk-libcxx-13.0.1/include to /nix/store/jgwba7i7c802n8l0mbhsmxx05br6j43d-libcxx-13.0.1-dev/include
mv: cannot move '/nix/store/v331jri254xij8a9fw8livmvzmfba1hk-libcxx-13.0.1/include/c++' to '/nix/store/jgwba7i7c802n8l0mbhsmxx05br6j43d-libcxx-13.0.1-dev/include/c++': Directory not empty
error: builder for '/nix/store/7gamcq7782qc7h23qvzf3rvnlvp2bmqf-libcxx-13.0.1.drv' failed with exit code 1;
       last 10 log lines:
       > -- Installing: /nix/store/v331jri254xij8a9fw8livmvzmfba1hk-libcxx-13.0.1/lib/libc++.1.0.dylib
       > -- Installing: /nix/store/v331jri254xij8a9fw8livmvzmfba1hk-libcxx-13.0.1/lib/libc++.1.dylib
       > -- Installing: /nix/store/v331jri254xij8a9fw8livmvzmfba1hk-libcxx-13.0.1/lib/libc++.dylib
       > -- Installing: /nix/store/v331jri254xij8a9fw8livmvzmfba1hk-libcxx-13.0.1/lib/libc++.a
       > /nix/store/21apr3mbmghnkd6zy2axxli2s36js3n8-clang-wrapper-11.1.0/bin/ranlib: file: /nix/store/v331jri254xij8a9fw8livmvzmfba1hk-libcxx-13.0.1/lib/libc++.a(functional.cpp.o) has no symbols
       > /nix/store/21apr3mbmghnkd6zy2axxli2s36js3n8-clang-wrapper-11.1.0/bin/ranlib: file: /nix/store/v331jri254xij8a9fw8livmvzmfba1hk-libcxx-13.0.1/lib/libc++.a(typeinfo.cpp.o) has no symbols
       > -- Installing: /nix/store/v331jri254xij8a9fw8livmvzmfba1hk-libcxx-13.0.1/lib/libc++experimental.a
       > post-installation fixup
       > Moving /nix/store/v331jri254xij8a9fw8livmvzmfba1hk-libcxx-13.0.1/include to /nix/store/jgwba7i7c802n8l0mbhsmxx05br6j43d-libcxx-13.0.1-dev/include
       > mv: cannot move '/nix/store/v331jri254xij8a9fw8livmvzmfba1hk-libcxx-13.0.1/include/c++' to '/nix/store/jgwba7i7c802n8l0mbhsmxx05br6j43d-libcxx-13.0.1-dev/include/c++': Directory not empty
       For full logs, run 'nix log /nix/store/7gamcq7782qc7h23qvzf3rvnlvp2bmqf-libcxx-13.0.1.drv'.
```

Reply:
Thanks, that's what I needed to know!